internal/regopolicyinterpreter/regopolicyinterpreter.go

```diff
@@ -329,16 +347,7 @@ func (r *RegoPolicyInterpreter) DisableLogging() error {
 	return nil
 }
 
-// Compile compiles the policy and its modules. This will increase the speed of policy
-// execution.
-func (r *RegoPolicyInterpreter) Compile() error {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
-
-	if r.compiledModules != nil {
-		return nil
-	}
-
+func (r *RegoPolicyInterpreter) compile() error {
```

Comment:
same as `updateMetadata`, can you add a comment saying `dataAndModulesMutex` needs to be held before calling?